src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -113,6 +114,15 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     SnapshotRequestModel snapshotReq =
         inputParameters.get(JobMapKeys.REQUEST.getKeyName(), SnapshotRequestModel.class);
     String snapshotName = snapshotReq.getName();
+    SnapshotRequestContentsModel contents = snapshotReq.getContents().get(0);
+    SnapshotRequestContentsModel.ModeEnum mode = contents.getMode();
+
+    // at start of flight, store the flight id in the snapshot request
+    if (mode == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
```

Comment:
Is there a specific reason we only want to do this in the `BYREQUESTID` case?

Reply:
It might be good to track the flight id for other modes maybe on the snapshotRequestModel (or some other place that makes sense), but currently, the byRequestId mode is the only mode whose snapshotRequestModel references a snapshotAccessRequest and that is what we are updating here. Because users will have the id from their snapshotAccessRequest and may care about this information, and by storing it on the snapshotAccessRequest we make it retrievable for the researcher.